code/gui/griddisplay.cpp

```diff
@@ -52,18 +68,62 @@ static void swap_rect_coords(int &x1, int &y1, int &x2, int &y2) {
     }
 }
 
+int GridDisplay::default_weight() {
+    return DEFAULT_WEIGHT;
+}
+
+int GridDisplay::grid_size() {
+    return GRID_SIZE;
+}
+
+double GridDisplay::scene_width() {
+    return m_scene->width();
+}
+
+double GridDisplay::scene_height() {
+    return m_scene->height();
+}
+
+double GridDisplay::parent_width() {
+    return m_image_viewer->width();
+}
+
+double GridDisplay::parent_height() {
+    return m_image_viewer->height();
+}
+
+double GridDisplay::x() {
+    return m_view->x();
+}
+
+double GridDisplay::y() {
+    return m_view->y();
+}
+
 GridDisplay::GridDisplay(ImageViewer *image_viewer, CameraDisplay *camera_display) :
     QWidget(image_viewer),
-    m_square_selected(40, 20),
+    m_image_viewer(image_viewer),
+
+    m_column_count(0),
+    m_row_count(0),
+
+    m_button(m_column_count, m_row_count),
+    m_square_selected(m_column_count, m_row_count),
+
     m_camera_display(camera_display) {
-    // Set up graphics scene and view
+
     m_scene = std::make_unique<QGraphicsScene>(this);
-    m_scene->setSceneRect(QRect(0, 0, SCENE_WIDTH, SCENE_HEIGHT));   //TODO: set bounding rectangle to m_img.size()
+    m_scene->setSceneRect(QRect(0, 0, SCENE_WIDTH, SCENE_HEIGHT));
+    
+    //TODO: set bounding rectangle to m_img.size()
+
     m_view = std::make_unique<QGraphicsView>(m_scene.get(), image_viewer);
     m_view->setStyleSheet("background: transparent");
-    m_view->setMinimumSize(SCENE_WIDTH, SCENE_HEIGHT);
+    m_view->setMinimumSize(0, 0);
```

Comment:
is this because `ImageViewer` might be resizing?

Reply:
Yeah if you set that then it can't resize below the minimum size I believe